test/e2e/multi-cluster-deployment/use-kubestellar.sh

```diff
@@ -51,6 +51,9 @@ kind: Deployment
 metadata:
   name: nginx-deployment
   namespace: nginx
+  annotations:
```

Comment:
Why did we decide that customization deserves to go into the shell testing?  Why not just add a ginkgo test for customization?

Reply:
I chose to put it there because I expect this will be commonly needed.

Reply:
I think having a specific test for that is better. I do agree we can add explicit section on customization in the user-guide which I think is what you did. 
I don't mind moving it out in future PR if you prefer not to spend time on that now, if so I'll open an issue for that.

Reply:
I think that adding a separate test just for template expansion is more costly to us contributors and more costly at runtime. I think it is better to fold that coverage into another test.